src/Libraries/RevitNodes/Elements/DetailCurve.cs

```diff
@@ -0,0 +1,160 @@
+    public class DetailCurve : CurveElement
+    {
+        #region Private constructors
+
+        /// <summary>
+        /// Construct a detail curve from an existing element
+        /// </summary>
+        /// <param name="element">Revit element</param>
+        private DetailCurve(Autodesk.Revit.DB.DetailCurve element)
+        {
+            SafeInit(() => InitElement(element));
+        }
+
+        /// <summary>
+        /// Construct a detail curve by curve
+        /// </summary>
+        /// <param name="view">View</param>
+        /// <param name="curve">Curve</param>
+        private DetailCurve(Autodesk.Revit.DB.View view, Autodesk.Revit.DB.Curve curve)
+        {
+            SafeInit(() => Init(view, curve));
+        }
+
+        #endregion
+
+        #region Helpers for private constructors
+
+        /// <summary>
+        /// Initialize Detail Curve element
+        /// </summary>
+        /// <param name="curve"></param>
+        private void InitElement(Autodesk.Revit.DB.DetailCurve curve)
+        {
+            InternalSetCurveElement(curve);
+        }
+
+        /// <summary>
+        /// Init a new detail curve from curve
+        /// </summary>
+        /// <param name="view"></param>
+        /// <param name="curve"></param>
+        private void Init(Autodesk.Revit.DB.View view, Autodesk.Revit.DB.Curve curve)
+        {
+            // Open Transaction and get document
+            Autodesk.Revit.DB.Document document = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+            
+            // Get exsiting element
+            var element = ElementBinder.GetElementFromTrace<Autodesk.Revit.DB.DetailCurve>(document);
+
+            if (element == null)
+            {
+                if (document.IsFamilyDocument)
+                {
+                    element = document.FamilyCreate.NewDetailCurve(view, curve);
+                }
+                else
+                {
+                    element = document.Create.NewDetailCurve(view, curve);
+                }                
+            }
+            else
+            {
+                element.SetGeometryCurve(curve, true);                
+            }
+
+            InternalSetCurveElement(element);
+
+            // Set transaction task done & element for trace
+            TransactionManager.Instance.TransactionTaskDone();
+            ElementBinder.SetElementForTrace(this.InternalElement);
+        }
+
+        #endregion
+
+        #region Public static constructors
+
+        /// <summary>
+        /// Construct a Revit DetailCurve element from a curve
+        /// </summary>
+        /// <param name="view">View to place the detail curve on</param>
+        /// <param name="curve">Curve to create detailcurve from</param>
+        /// <returns></returns>
+        public static DetailCurve ByCurve(Revit.Elements.Views.View view, Autodesk.DesignScript.Geometry.Curve curve)
```

Comment:
@mjkkirschner added planarity check, PolyCurve check and pulling it onto XY plane first now. There is no sketchplane needed for DetailCurves since they should be on XY.